docs/topics/development/waffle.md

```diff
@@ -0,0 +1,127 @@
+# Waffle
+
+We use [waffle](https://waffle.readthedocs.io/en/stable/) for managing feature access in production.
+
+## Why switches and not flags
+
+We prefer to use [switches](https://waffle.readthedocs.io/en/stable/types/switch.html)
+over flags, as switches are:
+
+- switches are simple
+- switches are easy to reason about
+
+## Creating/Deleting a switch
+
+Switches are added via database migrations.
+This ensures the switch exists in all environments once the migration is run.
+
+To create or remove a switch,
+first create an empty migration in the app where your switch will live.
+
+```bash
+python ./manage.py makemigrations <app> --empty
+```
+
+### Creating a switch
+
+add the switch in the migration
+
+```python
+# Generated by Django 4.2.8 on 2024-01-09 11:18
+
+from django.db import migrations
+
+from olympia.core.db.migrations import CreateWaffleSwitch
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('devhub', '0005_auto_20220531_1043'),
```

Comment:
I would not use a real app/migration name here (and drop the "Generated by..." comment above)

Reply:
(Same thing below)

Reply:
I compied the code generated by running the command, but happy to clean it up a bit.